### ISSUE TITLE
how is disable cache supposed to work?

### ISSUE THREAD

**Issue by Phylloxera** [apptainer/singularity#2635]
## Version of Singularity:

2.4.0

### Desired behavior

I want to pull a clean image from dockerhub each time (rather than from /root/.singularity/docker or from /root/.singularity/cache).
$`sudo rm -r singularity/diamond4/`
$`sudo rm -r /root/.singularity/docker/`
$`sudo mkdir /root/.singularity/docker/`
$`export SINGULARITY_DISABLE_CACHE=1`
$`sudo /usr/singularity2.4/bin/singularity build --sandbox singularity/diamond4/ docker://ubuntu`
Pulls a clean ubuntu from dockerhub

### Actual behavior

$`sudo rm -r singularity/diamond4/`
$`sudo rm -r /root/.singularity/docker/`
$`sudo mkdir /root/.singularity/docker/`
$`export SINGULARITY_DISABLE_CACHE=1`
$`sudo /usr/singularity2.4/bin/singularity build --sandbox singularity/diamond4/ docker://ubuntu`
Appears to be pulling the blobs I don't want from /root/.singularity/cache and putting them in /root/.singularity/docker and then building the image from that...
$`sudo /usr/singularity2.4/bin/singularity shell --writable singularity/diamond4/`
WARNING: Non existent bind point (file) in container: '/etc/localtime'
Singularity: Invoking an interactive shell within container...

Singularity diamond4:~>`ls`
LICENSE  diamond-0.9.24  v0.9.24.tar.gz  v0.9.24.tar.gz.1

### Related issues

I'm not entirely sure but think this might possibly relate some way to issue #1271 or issue #2519

**Comment by jscook2345** [comment]
@Phylloxera 

You should be able to verify if the cache is being used or not by adding the `-d` option to your commands.

Example:
<pre>
export SINGULARITY_DISABLE_CACHE=1
$sudo /usr/singularity2.4/bin/singularity -d build --sandbox singularity/diamond4/ docker://ubuntu
</pre>

**Comment by Phylloxera** [comment]
The result of the build command with -d is:
WARNING [U=0,P=80359]      sylabsToken()                 Authentication token file not found : Only pulls of public images will succeed
DEBUG   [U=0,P=80359]      func1()                       Registering plugin as type BuildPlugin
DEBUG   [U=0,P=80359]      NewBundle()                   Created temporary directory for bundle /tmp/sbuild-134065892
INFO    [U=0,P=80359]      Full()                        Starting build...
DEBUG   [U=0,P=80359]      updateCacheSubdir()           Caching directory set to /root/.singularity/cache/oci
Getting image source signatures
Skipping fetch of repeat blob sha256:38e2e6cd5626f31cea1a0a5751a9a7e6564c589a3388dcf84f00d4bb98146844
Skipping fetch of repeat blob sha256:705054bc3f5bd722eb8f026532447fac897c521d8906eb36e79a60d0fb0606fa
Skipping fetch of repeat blob sha256:c7051e0695642c5bf01467a4a64106cabfa20e57fee91b31ee73f7308bccf5bd
Skipping fetch of repeat blob sha256:7308e914506c09fa6d2242368545f55462e024f785e21b21b3e90403081a9336
Copying config sha256:fcef82905dbc8639dc501d9ed7c50e5b341b411b127f52c8735d24d10039cb51
 2.42 KiB / 2.42 KiB [======================================================] 0s
Writing manifest to image destination
Storing signatures
DEBUG   [U=0,P=80359]      func1()                       Running singularity_apps plugin: HandleBundle() hook
DEBUG   [U=0,P=80359]      BuildHandlePosts()            Running singularity_apps plugin: HandlePost() hook
DEBUG   [U=0,P=80359]      Full()                        Inserting Metadata
DEBUG   [U=0,P=80359]      Full()                        Calling assembler
INFO    [U=0,P=80359]      Assemble()                    Creating sandbox directory...
DEBUG   [U=0,P=80359]      Assemble()                    Moving sandbox from /tmp/sbuild-134065892/fs to singularity/diamond4/
INFO    [U=0,P=80359]      Full()                        Build complete: singularity/diamond4/

I'm not quite sure how to interpret the debug... Seems 'skipping fetch' means that as it is not going back out to the remote repository to get it from dockerhub... is that right? I think that those first four blobs are identical to what it pulls from dockerhub, which would mean all the extras I don't want are in 'config sha256:fcef82905dbc8639dc501d9ed7c50e5b341b411b127f52c8735d24d10039cb51' Not sure how I would go about finding and deleting this config... Am I getting warm? The resulting sandbox is the same on both v3.0 and 2.4.0 but on 2.4.0, I think the extras I don't want are in

VERBOSE2 Tar file with Docker env and labels: /root/.singularity/metadata/sha256:f6be9f4f6905406c1e7fd6031ee3104d25ad6a31d10d5e9192e7abf7a21e519a.tar.gz

But when I delete this file before a build, it is just recreated. I'm becoming more convinced this is user ignorance and not a technical issue with the program. I think I must have inadvertently created a config that is now adding this additional layer to all my ubuntu sandboxes that I don't want.

**Comment by jscook2345** [comment]
> I'm not quite sure how to interpret the debug... Seems 'skipping fetch' means that as it is not going back out to the remote repository to get it from dockerhub... is that right?

Yes. 'Skipping fetch of repeat blob...' means that the blob already exists in the cache and it is not going to retrieve it. The only way to have it fetch the new blobs is to delete the `$HOME/.singularity` cache each time before a build. If you're running as `root`, you will likely need to remove that `.singularity` cache as well

**Comment by Phylloxera** [comment]
I deleted everything I could find on the system with *singularity* or *diamond* in the name. Turns out some of my other singularity images were either corrupted or were being mounted in such a way that it appeared they were so I deleted all of these as well. Then I rebooted, did a fresh install of singularity 3.0 and everything seems to be back to normal. Still don't know what I did to cause that behavior but it appears to be fixed.
